platform/fabric/core/generic/channelprovider.go

```diff
@@ -28,41 +28,41 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-type VaultConstructor = func(configService driver.ConfigService, channel string, drivers []driver2.NamedDriver, metricsProvider metrics.Provider, tracerProvider trace.TracerProvider) (*vault.Vault, driver.TXIDStore, error)
+type VaultConstructor = func(
+	configService driver.ConfigService,
+	channel string,
+	drivers []driver2.NamedDriver,
+	metricsProvider metrics.Provider,
+	tracerProvider trace.TracerProvider,
+) (*vault.Vault, driver.TXIDStore, error)
 type LedgerConstructor func(
 	channelName string,
+	nw driver.FabricNetworkService,
```

Comment:
I don't know if it's a good idea to replace the actual dependencies by a struct that contains these and many more. This creates a tighter coupling and looser cohesion.